js/screens/SignUp/SignUpContainer.js

```diff
@@ -35,8 +46,27 @@ class SignUpContainer extends Component {
   }
 
   addUser() {
-    const { email, password, firstName, lastName, error } = this.props;
-    newUser(email, password, firstName, lastName, this.userError(error));
+    const validEmail = /^([\w-.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/;
+    const errorMessage = {
+      code: 'Invalid email or password',
+      message:
+        'Please Enter a valid email and password of at least 6 characters'
+    };
+    const userMessage = {
+      code: 'User already exists',
+      message: 'It looks like you already have an account!'
+    };
+    const { email, password, firstName, lastName, users } = this.props;
+
+    if (validEmail.test(email) && password.length > 5) {
```

Comment:
You can have this test as a helper function since you are using it more than once.